src/mrack/providers/openstack.py

```diff
@@ -301,7 +309,8 @@ def get_network(self, name=None, ref=None):
         network = self.networks.get(name)
         if not network:
             network = self.networks_by_ref.get(ref)
-        return network
+
+        return network if network is not None else {"name": name}
```

Comment:
As a user, I'd not expect this behavior from the `get_network` method. It's very "magicky" and thus I find it very problematic. Get method should return `None` if it doesn't exist, not some psedo object.